src/main/java/seedu/mtracker/console/EditInstrumentParser.java

```diff
@@ -75,52 +72,60 @@ public static void editReturnParameter(HashSet<String> parametersGiven) {
         if (!parametersGiven.contains(RETURN_ATTRIBUTE)) {
             return;
         }
-        TextUi.displayEditReturn();
-        String inputReturn = getUserInput(WORKSPACE);
-        if (!Validate.isValidPastReturns(inputReturn)) {
-            ErrorMessage.displayEditReturnError();
-            return;
-        }
+        String inputReturn;
+        do {
+            TextUi.displayEditReturn();
+            inputReturn = getUserInput(WORKSPACE);
+        } while (!Validate.isValidPastReturns(inputReturn));
         editedParameters.put(RETURN_ATTRIBUTE, inputReturn);
     }
 
     public static void editEntryPriceParameter(HashSet<String> parametersGiven) {
         if (!parametersGiven.contains(ENTRY_PRICE_ATTRIBUTE)) {
             return;
         }
-        TextUi.displayEditEntryPrice();
-        String inputEntryPrice = getUserInput(WORKSPACE);
-        if (!Validate.isValidPrice(inputEntryPrice)) {
-            ErrorMessage.displayEditPriceError();
-            return;
-        }
+        String inputEntryPrice;
+        do {
+            TextUi.displayEditEntryPrice();
+            inputEntryPrice = getUserInput(WORKSPACE);
+        } while (!Validate.isValidPrice(inputEntryPrice));
         editedParameters.put(ENTRY_PRICE_ATTRIBUTE, inputEntryPrice);
     }
 
     public static void editExitPriceParameter(HashSet<String> parametersGiven) {
         if (!parametersGiven.contains(EXIT_PRICE_ATTRIBUTE)) {
             return;
         }
-        TextUi.displayEditExitPrice();
-        String inputExitPrice = getUserInput(WORKSPACE);
-        if (!Validate.isValidPrice(inputExitPrice)) {
-            ErrorMessage.displayEditPriceError();
-            return;
-        }
+        String inputExitPrice;
+        do {
+            TextUi.displayEditExitPrice();
+            inputExitPrice = getUserInput(WORKSPACE);
+        } while (!Validate.isValidPrice(inputExitPrice));
         editedParameters.put(EXIT_PRICE_ATTRIBUTE, inputExitPrice);
     }
 
     public static void editExpiryParameter(HashSet<String> parametersGiven) {
         if (!parametersGiven.contains(EXPIRY_ATTRIBUTE)) {
             return;
         }
-        TextUi.displayEditExpiry();
-        String inputExpiry = getUserInput(WORKSPACE);
-        if (!Validate.isValidExpiry(inputExpiry)) {
-            ErrorMessage.displayEditExpiryError();
+        String inputExpiry;
+        do {
+            TextUi.displayEditExpiry();
+            inputExpiry = getUserInput(WORKSPACE);
+        } while (!Validate.isValidExpiry(inputExpiry));
+        editedParameters.put(EXPIRY_ATTRIBUTE, inputExpiry);
+    }
+
+    public static void editDoneStatus(HashSet<String> parametersGiven) {
+        if (!parametersGiven.contains(DONE_ATTRIBUTE)) {
             return;
         }
-        editedParameters.put(EXPIRY_ATTRIBUTE, inputExpiry);
+        String inputStatus;
+        do {
+            TextUi.displayEditStatus();
+            inputStatus = getUserInput(WORKSPACE);
```

Comment:
make it to lowercase so that it is case insensitive since we only allow 2 values done and undone